src/core/api/base.go

```diff
@@ -217,6 +223,16 @@ func Init() error {
 	}
 	err = scheduler.RegisterCallbackFunc(preheat.SchedulerCallback, p2pPreheatCallbackFun)
 
+	gcCallbackFun := func(ctx context.Context, p string) error {
```

Comment:
Check error for the register of p2p callback function in line 224

Reply:
And suggest to move the registering logic into the package of gc `controller/gc`. After the refactoring in the future, the current file should be removed